src/main/java/si/mazi/rescu/clients/HttpConnection.java

```diff
@@ -0,0 +1,56 @@
+package si.mazi.rescu.clients;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.ProtocolException;
+import java.util.List;
+import java.util.Map;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLSocketFactory;
+
+import si.mazi.rescu.HttpMethod;
+
+public interface HttpConnection {
+
+    HttpConnectionType getHttpConnectionType();
+    String getHeaderField(String name);
+    String getRequestMethod();
+    OutputStream getOutputStream() throws IOException;
+    int getResponseCode() throws IOException;
+    Map<String,List<String>> getHeaderFields();
+    InputStream getInputStream() throws IOException;
+    InputStream getErrorStream() throws IOException;
+    void setRequestMethod(HttpMethod method) throws ProtocolException;
+    void addHeader(String key, String value);
+    void setDoOutput(boolean b);
```

Comment:
Please remove the setDoOutput and setDoInput methods and replace them with:
void doFinalConfig(Integer contentLength);
Its Apache (or default) implementation should be empty, Java.impl should have:

            connection.setDoOutput(true);
            connection.setDoInput(true);